pkg/network/netlink/conntracker.go

```diff
@@ -88,13 +102,22 @@ func NewConntracker(config *config.Config) (Conntracker, error) {
 func newConntrackerOnce(procRoot string, maxStateSize, targetRateLimit int, listenAllNamespaces bool) (Conntracker, error) {
 	consumer := NewConsumer(procRoot, targetRateLimit, listenAllNamespaces)
 	ctr := &realConntracker{
-		consumer:             consumer,
-		compactTicker:        time.NewTicker(compactInterval),
-		state:                make(map[connKey]*network.IPTranslation),
-		maxStateSize:         maxStateSize,
-		exceededSizeLogLimit: util.NewLogLimit(10, time.Minute*10),
+		consumer:      consumer,
+		orphans:       list.New(),
+		maxStateSize:  maxStateSize,
+		compactTicker: time.NewTicker(compactInterval),
+		orphanTimeout: defaultOrphanTimeout,
 	}
 
+	ctr.cache, _ = simplelru.NewLRU(maxStateSize, func(key, value interface{}) {
+		// ctr lock should be held, so it is
+		// safe to modify ctr
+		t := value.(*translationEntry)
+		if t.orphan != nil {
+			ctr.orphans.Remove(t.orphan)
+		}
```

Comment:
sorry, I'm bit confused by this comment. Are you saying that the go-routine that processes the eviction callback is the same one adding an entry to the LRU cache (and thus holding the mutex?)

Reply:
Whenever any operation is done on the cache, the lock is held. The evict callback is called during an add or remove operation on the cache, both times we hold the lock (the evict callback is called synchronously).

Reply:
gotcha. thanks 👍